### PR TITLE
7 voting

### PR DESCRIPTION
**Pull request by NRMSMITH**

### CODE REVIEW

File: src/Pages/SingleArticle.jsx [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
     const [singleArticle, setSingleArticle] = useState([]);
     const [isLoading, setIsLoading] = useState(false)
+    const [count, setCount] = useState(0);
     const { article_id } = useParams();
 
+    const changeVote = () => {
+        updateVote(article_id)
+        setCount(count + 1)
+    }
+
```
Comment (popatre) [inline src/Pages/SingleArticle.jsx]: Just needs some error handling incase it goes wrong. 
At the moment, i can click the button even when offline, and it looks like its going up, but isnt.

**Review by popatre — CHANGES_REQUESTED** [review]
just missing some error handling for if the request doesnt work

**Review by popatre — APPROVED** [review]

File: src/Pages/SingleArticle.jsx [hunk 2]
```diff
@@ -7,10 +7,16 @@ export const SingleArticle = () => {
...
+    const [err, setErr] = useState(null);
     const { article_id } = useParams();
 
     const changeVote = () => {
+        setErr(null)
         updateVote(article_id)
+        .catch((err) => {
+            setCount((count) => count - 1);
+            setErr('Something went wrong, please try again.')
+        })
```
Comment (popatre) [inline src/Pages/SingleArticle.jsx:19]: I would jsut move this to above the request

**Review by popatre — APPROVED** [review]

File: src/Pages/SingleArticle.jsx (hunk 2 above)
Comment (NRMSMITH) [reply src/Pages/SingleArticle.jsx:19]: So just above the updateVote that links to the api request?

**Pull request merged**